### ISSUE TITLE
KeeLoq: Supported manufacturers

### ISSUE THREAD

**Issue by Stichoza** [DarkFlippers/unleashed-firmware#269]
### Describe the enhancement you're suggesting.

Hello,

I'd love to help with improving support of additional KeeLoq manufacturers.

I have multiple `.sub` files recorded from Read mode with Flipper that shows up as `KL Unknown` (KeeLoq 64bit with key, fix, hop and button code, but without count and manufacturer).

Will it help the development process if I provide those sub files? Or anything else I can do?

### Anything else?

Also I've seen [here](https://www.ifoedit.com/garagehack.html) (under detected/created/copied) that there are some devices that are learnable, but have no Flipper support. Is that a hardware limitation?

<img width="480" alt="image" src="https://user-images.githubusercontent.com/1139050/211060544-98146553-9244-4453-8f76-cc12ab603dc9.png">

P.S.Huge thanks for developing this firmware, it's the best of all FWs!

**Comment by xMasterX** [comment]
Screenshot you provided is not related to our firmware in any case, this is someone’s commercial project and it says about his support, not ours

Attach your sub files here ill take a look on them

**Comment by Stichoza** [comment]
Yes, I know that's a totally different thing on the screenshot. 

### Recorded signals
Here is a link to a gist containing captured signals: https://gist.github.com/Stichoza/fc61b41a4c5ae49c1fe59effd78c9713

It contains RAW recording of multiple presses per button, as well as decoded signals (Decoded as KL Unknown).

### Remote
Remote has three buttons and looks like this: 
![images](https://user-images.githubusercontent.com/1139050/214048548-3659e22e-2600-41e8-bdea-b8eb03c110cc.jpeg)

**Comment by LY2NEO** [comment]
... just want to add, that some manufacturer keys which are already on flipper are partially decrypting where decrypted part of serial number are zeros, that gives oportunity to decrypt and see counter, but in revers when encrypting plain we getting wrong chipper and generated next hop is wrong and not working... I think it is due to not play everybody with neighbor garage doors 😀 but with power of math and powerful GPU I thing it is possible to get rigth ones 😉

**Comment by calb33** [comment]
@xMasterX Just wondering if I can bother you with this one? It's a Keeloq remote called PROKEY2 from Australian company called CSD that is not detected yet on Unleashed FW. It runs on 433.92 and 650AM (even though it reads a bit funny on Freq Analyzer, jumps around 433.89 - 433.91). I've attached the RAW Read (renamed the sub into txt) from 10 button presses and photo of the PCB inside the remote (no FCC label unfortunately). Thanks! 😸 
![333598427_629300112354002_598833998020670468_n - Copy](https://user-images.githubusercontent.com/128198291/226076494-7912ba7f-4e4d-4960-8c9b-4ed203a1ca87.jpg)

[RAW_ButtonB.txt](https://github.com/DarkFlippers/unleashed-firmware/files/11007214/RAW_ButtonB.txt)

**Comment by realcatgirly** [comment]
Hi, I'd also like to contribute some signals. I have a remote that decodes as "KL Unknown", the remote looks like a "Novoferm MNHS", but has a different branding on it ("Siebau", but seems to be just a branding perhaps)

**Comment by xMasterX** [comment]
> Hi, I'd also like to contribute some signals. I have a remote that decodes as "KL Unknown", the remote looks like a "Novoferm MNHS", but has a different branding on it ("Siebau", but seems to be just a branding perhaps)

Can you share RAW recordings of this remote??
3 presses of each button with pause around 2 seconds between each press, you can record them in one raw file

**Comment by mishamyte** [comment]
Hi @Stichoza!

Thanks for you proposal. 
So far over years, we extended our support for some KL-based protocols.

So you can check it on latest builds.
